### ISSUE TITLE
PGA memory leak with LOBs (Regression)

### ISSUE THREAD

**Issue by nenadnoveljic** [sijms/go-ora#477]
The [memory leak bug](https://github.com/sijms/go-ora/issues/439) that was fixed once in the release `2.7.6` started appearing again in `2.7.24`. Actually, the bug was reintroduced in `2.7.7`

**Comment by sijms** [comment]
result after use sqlx
```bash
PGA [bytes]: 8906750.000000 
Temp LOB count: 1 
PGA [bytes]: 8879358.000000 
Temp LOB count: 2 
PGA [bytes]: 8879358.000000 
Temp LOB count: 3 
PGA [bytes]: 8879358.000000 
Temp LOB count: 4 
PGA [bytes]: 8879358.000000 
Temp LOB count: 5 
PGA [bytes]: 8879358.000000 
Temp LOB count: 6 
PGA [bytes]: 8879358.000000 
Temp LOB count: 7 
PGA [bytes]: 8879358.000000 
Temp LOB count: 8 
PGA [bytes]: 8965206.000000 
Temp LOB count: 9 
PGA [bytes]: 8965206.000000 
Temp LOB count: 10 
```

**Comment by nenadnoveljic** [comment]
Both test cases are showing the leak, as the number of temp LOB locators is always increasing. 

Do you agree?

Just, with sqlx I would expect that the PGA is also increasing.

Have you also replaced `sql` in the test case?

Without sqlx it's the PL/SQL anonymous block: `sql := "BEGIN SELECT to_clob('A') into :c FROM dual; END;`

With sqx it has to be a query `sql := "SELECT to_clob('A') FROM dual"`

**Comment by sijms** [comment]
result after correction by calling lob.close(0x10000)
```bash
PGA [bytes]: 8906750.000000 
Temp LOB count: 1 
PGA [bytes]: 8899838.000000 
Temp LOB count: 2 
PGA [bytes]: 8886550.000000 
Temp LOB count: 0 
PGA [bytes]: 8879086.000000 
Temp LOB count: 1 
PGA [bytes]: 8879086.000000 
Temp LOB count: 2 
PGA [bytes]: 8886550.000000 
Temp LOB count: 0 
PGA [bytes]: 8879086.000000 
Temp LOB count: 1 
PGA [bytes]: 8899566.000000 
Temp LOB count: 2 
PGA [bytes]: 8886550.000000 
Temp LOB count: 0 
PGA [bytes]: 8879086.000000 
Temp LOB count: 1 
```
the problem now is there is some sort of lab introduce by calling this function for each lob data

**Comment by sijms** [comment]
I have function in `command.go` which free all temp lobs at the end of `query/exec`  it will help if you have multiple lobs in same query/exec call.

so I will update the code by adding all these lob locator to temp lob array of the command and let freeTemporayLobs do its job

**Comment by sijms** [comment]
result after add returned lobs to stmt.temporayLobs
```bash
PGA [bytes]: 8906750.000000 
Temp LOB count: 0 
PGA [bytes]: 8879358.000000 
Temp LOB count: 0 
PGA [bytes]: 8879358.000000 
Temp LOB count: 0 
PGA [bytes]: 8879358.000000 
Temp LOB count: 0 
PGA [bytes]: 8879358.000000 
Temp LOB count: 0 
PGA [bytes]: 8899838.000000 
Temp LOB count: 0 
PGA [bytes]: 8899838.000000 
Temp LOB count: 0 
PGA [bytes]: 8899838.000000 
Temp LOB count: 0 
PGA [bytes]: 8899838.000000 
Temp LOB count: 0 
PGA [bytes]: 8899838.000000 
Temp LOB count: 0 
```

**Comment by sijms** [comment]
I make a new commit would you please test it before make new release

**Comment by sijms** [comment]
fixed in v2.7.26
